### PR TITLE
Upgrade to Wagtail 2.0.1 and Django 1.11.11

### PR DESCRIPTION
**Pull request by patphongs**
## Summary
- Addresses #1538 
_This upgrades Wagtail to version 2.0.1 and Django 1.11.11. This includes updates to the Wagtail CMS interface and rich text editor with more user-friendly functionality. Also includes additional `Glossary` button for adding glossary terms in Wagtail._

## Impacted areas of the application
List general components of the application that this PR will affect:

- Wagtail

### Rich text editor options
screen shot 2018-05-02 at 1 18 42 pm

### CODE REVIEW

**Comment by patphongs** [comment]
@annalee @rjayasekera After working on this issue with you, below I believe is the solution that we came up with. We'll need to make sure that the following instructions work. Also this PR has a dependency https://github.com/fecgov/fec-cms/issues/2006, so we SHOULD NOT merge this wagtail and django upgrade until that issue is closed:

## Deploy to cloud environments

### Step 1
```
# Updated requirements for Django==1.11.13 and wagtail==2.0.1. Run new install of requirements
pip install -r requirements.txt
# Disabled the django tests in circle for this PR
# Run all the wagtail app migrations first on deploy. We put all of these dependencies in the initial migrations so that the tests will pass.
./manage.py migrate wagtailadmin
./manage.py migrate wagtailcore
./manage.py migrate wagtaildocs
./manage.py migrate wagtailembeds
./manage.py migrate wagtailforms
./manage.py migrate wagtailimages
./manage.py migrate wagtailredirects
./manage.py migrate wagtailsearch
./manage.py migrate wagtailsearchpromotions
./manage.py migrate wagtailusers
# Run all the other migrations
./manage.py migrate
# Deploy will run and should deploy without issue
```

### Step 2
```
# Re-enable django tests in circle.yml
# Update 0001_initial.py with wagtail 0040_page_draft_title migration to make sure that runs first before running home for tests to pass. A separate PR should be submitted to add this dependency, so that it does not run on first deploy.
    dependencies = [
        ('wagtailcore', '0040_page_draft_title'),
    ]

# With this change, tests should now pass and deploy should still go as expected since the initial deploy should take care of the database changes on deploy
```
## Local set up to pass tests and runserver
```
# Checkout this branch: feature/1538-upgrade-wagtail-2.0-django-1.11
# Run new install of requirements
pip install -r requirements.txt
# Run all the wagtail app migrations first.
./manage.py migrate wagtailadmin
./manage.py migrate wagtailcore
./manage.py migrate wagtaildocs
./manage.py migrate wagtailembeds
./manage.py migrate wagtailforms
./manage.py migrate wagtailimages
./manage.py migrate wagtailredirects
./manage.py migrate wagtailsearch
./manage.py migrate wagtailsearchpromotions
./manage.py migrate wagtailusers
# Run all the other migrations
./manage.py migrate
# Check that all migrations applied
./manage.py showmigrations
# Run the server to see if the server starts up ok
./manage.py runserver
# After confirming that CMS starts up ok, update the dependencies to include the wagtailcore app dependency to the 0001_initial.py migration file
    dependencies = [
        ('wagtailcore', '0040_page_draft_title')
    ]
# Run the tests
./manage.py test
# Run the server
./manage.py runserver
```

File: fec/fec/draftail/anchor.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        'data-anchor': props['anchor'],
+        'id': props['anchor'],
+        'style': {
+            'fontSize': '2.4rem',
+            'margin': '0 0 1em 0',
+            'fontFamily': 'gandhi,serif',
+            'fontSize': '2rem',
+            'fontWeight': '700',
+            'lineHeight': '1.2'
+        } ,
```
[the remark this answers is not in the document]
Comment (apburnes) [reply fec/fec/draftail/anchor.py]: Great point! The `h2` was not being assigned the `data-term` attributes but the `span` would take them. We could try appending the h2 anchor in the [javascript function](https://github.com/fecgov/fec-cms/blob/0bd83b2e215c16f942effe50a4d5fb1507aa58d7/fec/fec/static/js/modules/helpers.js#L38-L49) or make it a child element.

**Comment by dorothyyeager** [comment]
@apburnes asked me to mention this here: It would be desirable for some new pages under development if the "related topics" block could be available as an option (above citations block) in the custom page template.

**Comment by apburnes** [comment]
@dorothyyeager This is `related_topics` addition to the _Custom Page_ template
screen shot 2018-06-29 at 2 49 56 pm

How's this?

**Comment by dorothyyeager** [comment]
@apburnes That looks perfect and will do what we are looking for it to do for these new pages. Thanks!

**Pull request merged**